mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -563,7 +563,16 @@ private[ml] class DefaultParamsReader[T] extends MLReader[T] {
 
   override def load(path: String): T = {
     val metadata = DefaultParamsReader.loadMetadata(path, sparkSession)
-    val cls = Utils.classForName(metadata.className)
+    val cls = try {
+      // Only loads whitelisted classes. This is for preventing RCE in Spark Connect mode.
+      val MLHandlerClazz = Utils.classForName("org.apache.spark.sql.connect.ml.MLHandler")
+      val safeMLClassLoader = MLHandlerClazz.getMethod("safeMLClassLoader").invoke(null)
```

Comment:
should it be a class variable?

Reply:
it should be a function that returns the loaded class.